core/src/test/java/org/apache/iceberg/view/TestViewMetadata.java

```diff
@@ -0,0 +1,280 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.view;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.types.Types;
+import org.junit.jupiter.api.Test;
+
+public class TestViewMetadata {
+
+  @Test
+  public void nullAndMissingFields() {
+    assertThatThrownBy(() -> ImmutableViewMetadata.builder().build())
+        .isInstanceOf(IllegalStateException.class)
+        .hasMessage(
+            "Cannot build ViewMetadata, some of required attributes are not set [formatVersion, location, currentSchemaId, currentVersionId]");
+
+    assertThatThrownBy(() -> ImmutableViewMetadata.builder().formatVersion(1).build())
+        .isInstanceOf(IllegalStateException.class)
+        .hasMessage(
+            "Cannot build ViewMetadata, some of required attributes are not set [location, currentSchemaId, currentVersionId]");
+
+    assertThatThrownBy(
+            () -> ImmutableViewMetadata.builder().formatVersion(1).location("location").build())
+        .isInstanceOf(IllegalStateException.class)
+        .hasMessage(
+            "Cannot build ViewMetadata, some of required attributes are not set [currentSchemaId, currentVersionId]");
+
+    assertThatThrownBy(
+            () ->
+                ImmutableViewMetadata.builder()
+                    .formatVersion(1)
+                    .location("location")
+                    .currentSchemaId(1)
+                    .build())
+        .isInstanceOf(IllegalStateException.class)
+        .hasMessage(
+            "Cannot build ViewMetadata, some of required attributes are not set [currentVersionId]");
+
+    assertThatThrownBy(
+            () ->
+                ImmutableViewMetadata.builder()
+                    .formatVersion(1)
+                    .location("location")
+                    .currentSchemaId(1)
+                    .currentVersionId(1)
+                    .build())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Invalid view versions: empty");
+  }
+
+  @Test
+  public void unsupportedFormatVersion() {
+    assertThatThrownBy(
+            () ->
+                ImmutableViewMetadata.builder()
+                    .formatVersion(23)
```

Comment:
It would be better to test 2, since that's the next version that would be added.

Reply:
makes sense, updated